addons/test_mail/tests/test_message_post.py

```diff
@@ -211,115 +241,308 @@ def test_notify_recipients_internals(self):
                 self.assertFalse(partner_info['has_button_access'])
                 self.assertFalse(emp_info['has_button_access'])
 
+
+@tagged('mail_post', 'mail_notify')
+class TestMessageNotify(TestMessagePostCommon):
+
+    @users('employee')
     @mute_logger('odoo.addons.mail.models.mail_mail')
-    def test_post_needaction(self):
-        (self.user_employee | self.user_admin).write({'notification_type': 'inbox'})
-        with self.assertSinglePostNotifications([{'partner': self.partner_employee, 'type': 'inbox'}], {'content': 'Body'}):
-            self.test_record.message_post(
-                body='Body', message_type='comment', subtype_xmlid='mail.mt_comment',
-                partner_ids=[self.user_employee.partner_id.id])
-
-        self.test_record.message_subscribe([self.partner_1.id])
-        with self.assertSinglePostNotifications([
-                {'partner': self.partner_employee, 'type': 'inbox'},
-                {'partner': self.partner_1, 'type': 'email'}], {'content': 'NewBody'}):
-            self.test_record.message_post(
-                body='NewBody', message_type='comment', subtype_xmlid='mail.mt_comment',
-                partner_ids=[self.user_employee.partner_id.id])
+    def test_notify(self):
+        test_record = self.env['mail.test.simple'].browse(self.test_record.ids)
+
+        with self.mock_mail_gateway():
+            new_notification = test_record.message_notify(
+                body='<p>You have received a notification</p>',
+                partner_ids=[self.partner_1.id, self.partner_admin.id, self.partner_employee_2.id],
+                subject='This should be a subject',
+            )
+
+        self.assertMessageFields(
+            new_notification,
+            {'author_id': self.partner_employee,
+             'body': '<p>You have received a notification</p>',
+             'email_from': formataddr((self.partner_employee.name, self.partner_employee.email_normalized)),
+             'message_type': 'user_notification',
+             'model': test_record._name,
+             'notified_partner_ids': self.partner_1 | self.partner_employee_2 | self.partner_admin,
+             'res_id': test_record.id,
+             'subtype_id': self.env.ref('mail.mt_note'),
+            }
+        )
+        self.assertNotIn(new_notification, self.test_record.message_ids)
+
+        admin_mails = [x for x in self._mails if self.partner_admin.name in x.get('email_to')[0]]
+        self.assertEqual(len(admin_mails), 1, 'There should be exactly one email sent to admin')
+        admin_mail = admin_mails[0].get('body')
+        admin_access_link = admin_mail[admin_mail.index('model='):admin_mail.index('/>') - 1] if 'model=' in admin_mail else None
```

Comment:
a regex is a bit more explicit imo, as you prefer
```suggestion
        admin_access_link = re.search(r'model=[\w]+', admin_mail).group()
```

Reply:
Actually due to the res_mode=..&res_id=.. I have to make several re or a complex one. I prefer plain stupid indexes :D .